File: Digito Verificador/Modulo 11 Padrao/DV Gerador/gerardig_mod11.py
def validarMatricula(MATRICULA):
    if len(MATRICULA) != 7:
        return False
    if not MATRICULA[0].isdigit() or not MATRICULA[6].isdigit():
        return False
    if MATRICULA[3] != '.':
        return False
    for i in range(1, 3):
        if not MATRICULA[i].isdigit() or not MATRICULA[i + 4].isdigit():
            return False
    return True

def removerCaractere(CADEIA, CARACTERE):
    return CADEIA.replace(CARACTERE, "")

def mod11(MATRIC_CPT):
    MATRIC_SPT = removerCaractere(MATRIC_CPT, ".")
    MATRIC_INT = [int(char) for char in MATRIC_SPT]
    SOMA = sum(MATRIC_INT[i] * (7 - i) for i in range(6))
    if SOMA % 11 == 0:
        DV = "0"
    elif SOMA % 11 == 1:
        DV = "X"
    elif SOMA % 11 == 10:
        DV = "1"
    else:
        DV = str(11 - (SOMA % 11))
    return DV

def main():
    while True:
        MATRICULA = input("Informe matricula no formato 999.999: ")
        if not validarMatricula(MATRICULA):
            print()
            print("Matricula invalida. Tente novamente.")
            print()
        else:
            break

    DV = mod11(MATRICULA)
    print()
    print(f"Matricula com DV = {MATRICULA}-{DV}")
    print()
    input("Tecle <Enter> para encerrar...")

if __name__ == "__main__":
    main()

File: Digito Verificador/Modulo 11 Padrao/DV Gerador/gerardig_mod11.java
import java.util.Scanner;

public class Main {
    public static boolean validarMatricula(String MATRICULA) {
        if (MATRICULA.length() != 7)
            return false;
        if (!Character.isDigit(MATRICULA.charAt(0)) || !Character.isDigit(MATRICULA.charAt(6)))
            return false;
        if (MATRICULA.charAt(3) != '.')
            return false;
        for (int I = 1; I <= 2; I++)
            if (!Character.isDigit(MATRICULA.charAt(I)) || !Character.isDigit(MATRICULA.charAt(I + 4)))
                return false;
        return true;
    }

    public static String removerCaractere(String CADEIA, char CARACTERE) {
        return CADEIA.replace(Character.toString(CARACTERE), "");
    }

    public static char mod11(String MATRIC_CPT) {
        String MATRIC_SPT = MATRIC_CPT;
        MATRIC_SPT = removerCaractere(MATRIC_SPT, '.');
        int[] MATRIC_INT = new int[6];
        for (int I = 0; I <= 5; I++)
            MATRIC_INT[I] = MATRIC_SPT.charAt(I) - 48;
        int SOMA = 0;
        for (int I = 0; I <= 5; I++)
            SOMA += MATRIC_INT[I] * (7 - I);
        char DV;
        if ((SOMA % 11) == 0)
            DV = '0';
        else if ((SOMA % 11) == 1)
            DV = 'X';
        else if ((SOMA % 11) == 10)
            DV = '1';
        else
            DV = (char) ((11 - (SOMA % 11)) + 48);
        return DV;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String MATRICULA;
        char DV;

        do {
            System.out.print("Informe matricula no formato 999.999: ");
            MATRICULA = scanner.nextLine();
            if (!validarMatricula(MATRICULA)) {
                System.out.println();
                System.out.println("Matricula invalida. Tente novamente.");
                System.out.println();
            }
        } while (!validarMatricula(MATRICULA));

        DV = mod11(MATRICULA);
        System.out.println();
        System.out.printf("Matricula com DV = %s-%c%n", MATRICULA, DV);
        System.out.println();
        System.out.print("Tecle <Enter> para encerrar... ");
        scanner.nextLine();
    }
}
